src/Akeneo/Pim/Enrichment/Component/Product/Model/QuantifiedAssociation/QuantifiedAssociations.php

```diff
@@ -129,6 +129,38 @@ public function getQuantifiedAssociationsProductModelCodes(): array
         return $result;
     }
 
+    public function merge(QuantifiedAssociations $quantifiedAssociations): void
+    {
+        $normalizedQuantifiedAssociationsToMerge = $quantifiedAssociations->normalize();
+
+        foreach ($normalizedQuantifiedAssociationsToMerge as $associationTypeCode => $association) {
+            foreach ($association as $associationEntityType => $quantifiedLinks) {
+                if (!isset($this->quantifiedAssociations[$associationTypeCode][$associationEntityType])) {
+                    $this->quantifiedAssociations[$associationTypeCode][$associationEntityType] = [];
+                }
+
+                $entityQuantifiedLinks = $this->quantifiedAssociations[$associationTypeCode][$associationEntityType];
+                foreach ($quantifiedLinks as $quantifiedLink) {
+                    $key = $this->searchKeyOfDuplicatedQuantifiedAssociation(
+                        $this->quantifiedAssociations,
+                        $associationTypeCode,
+                        $associationEntityType,
+                        $quantifiedLink
+                    );
+
+                    if (null !== $key) {
+                        $this->quantifiedAssociations[$associationTypeCode][$associationEntityType][$key] = new QuantifiedLink(
+                            $this->quantifiedAssociations[$associationTypeCode][$associationEntityType][$key]->identifier(),
+                            $quantifiedLink['quantity']
+                        );
+                    } else {
+                        $this->quantifiedAssociations[$associationTypeCode][$associationEntityType][] = new QuantifiedLink($quantifiedLink['identifier'], $quantifiedLink['quantity']);
```

Comment:
```suggestion
                        $this->quantifiedAssociations[$associationTypeCode][$associationEntityType][] = new QuantifiedLink(
                            $quantifiedLink['identifier'],
                            $quantifiedLink['quantity']
                        );
```

I know it's only code style, but it's easier to read and is consistent with the similar line just before.

Reply:
btw, what's the point of creating a new QuantifiedLink with the exact same values ?
In the initial version, it was `$quantifiedAssociations[$associationTypeCode][$associationEntityType][] = $quantifiedLink;`
What was the issue with this version ?